kore/src/Kore/Step/Simplification/Predicate.hs

```diff
@@ -48,38 +44,23 @@ simplifyPartial
     => PredicateSubstitutionSimplifier level
     -> StepPatternSimplifier level
     -> Predicate level variable
-    -> Simplifier
-        ( PredicateSubstitution level variable
-        , SimplificationProof level
-        )
+    -> BranchT Simplifier (PredicateSubstitution level variable)
 simplifyPartial
     substitutionSimplifier
     (StepPatternSimplifier simplifier)
     predicate
   = do
     (patternOr, _proof) <-
-        simplifier substitutionSimplifier (unwrapPredicate predicate)
-    case MultiOr.extractPatterns patternOr of
-        [] -> return
-            ( Predicated.bottomPredicate
-            , SimplificationProof
-            )
-        [ Predicated
-                { term = Top_ _
-                , predicate = simplifiedPredicate
-                , substitution = simplifiedSubstitution
-                }
-            ] -> return
-                ( Predicated
-                    { term = ()
-                    , predicate = simplifiedPredicate
-                    , substitution = simplifiedSubstitution
-                    }
-                , SimplificationProof
-                )
-        [patt] ->
-            (error . show . Pretty.vsep)
-                [ "Expecting a top term!"
-                , unparse patt
-                ]
-        _ -> error ("Expecting at most one result " ++ show patternOr)
+        Monad.Trans.lift
```

Comment:
Shouldn't this be `liftPruned`?

Reply:
It would be redundant, because we `scatter` the result below. If the `OrOfExpandedPattern` is empty, then `scatter`-ing it is the same as returning `empty` here. I made a note of this below.